src/cpp/shared_core/include/shared_core/Logger.hpp

```diff
@@ -92,7 +92,8 @@ enum class LogLevel
    ERR = 1,       // Error messages will be logged.
    WARN = 2,      // Warning and error messages will be logged.
    INFO = 3,      // Info, warning, and error messages will be logged.
-   DEBUG = 4      // All messages will be logged.
+   DEBUG = 4,     // All messages will be logged.
+   DEBUG_LEVEL = 4// All messages will be logged - use to avoid name conflict with DEBUG macro
```

Comment:
I don't understand the need for `DEBUG` and `DEBUG_LEVEL`.

Reply:
In Macros.hpp we have a `#define DEBUG` macro that's used all over the place in rsession. So if you include that header and Logger.hpp you can't refer to LogLevel::DEBUG without a compile error.  Since I needed to use that constant from LOG_DEBUG_MESSAGE, I needed a constant that would not conflict.

Reply:
I don't love this as it seems like a pollution of the interface to get around the `rsession` specific code. Is it possible we can tweak the DEBUG macro? If not, no worries.

Reply:
It might make sense to rename that macro to something less likely to collide, like DBG?

Reply:
What does the macro do? Could we give it a more descriptive name, like "ONLY_IN_DEBUG" or something?

Reply:
I originally went and changed ~20 files from using DEBUG(...) to RSTUDIO_DEBUG(...) but it felt like overkill for a name conflict on a symbol like this that's not useful very often.  Those are valuable messages for debugging details of various components like RMarkdown, and other stuff in rsession but pretty sure they are compiled out by default.